### ISSUE TITLE
`LookupRecord` (and LookupMany) aliases

### ISSUE THREAD

**Issue by combinatorist** [gristlabs/grist-core#584]
I just filed ticket #583 because I couldn't figure out why I couldn't use: `myTable.lookupRecord(...`. I assumed it was because it was a summary table, but in fact, it's just because there is no alias for singular record lookups. I'd vote to resolve this inconsistent terminology, probably by completing the options, so there is a singular complement to `lookupRecords` and plural complement to `lookupOne`. I don't know how many times that's tripped me up and now that I've been busy with other things, it tripped me up again as soon as I tried to write a singular lookup function again.

**Comment by dsagal** [comment]
I see no problem with adding a singular `lookupRecord` as an alias to `lookupOne`.

Curious what you think of another alternative: what if looking up one record were available as `myTable.lookupRecods(....)[0]` (while remaining as efficient as `lookupOne`)? Would that be convenient, in terms of fewer method names to remember?

**Comment by combinatorist** [comment]
> Curious what you think of another alternative: what if looking up one record were available as `myTable.lookupRecods(....)[0]` (while remaining as efficient as `lookupOne`)? Would that be convenient, in terms of fewer method names to remember?

If I were designing your API, I would be tempted to eventually consolidate to a single general pattern, but I like the explicit methods for finding one or many, although I suppose that's because I expect the singular method to raise an error if it finds 0 or many and it doesn't actually do that anyway. Right now, it's probably best described as a lookupFirstOption (in the scala sense of an Option). So, yeah, your indexed solution is at least honest that it's the first one (and makes it almost as easy to take the last if someone wants that or anything else more general). I think it might turn off some people less familiar with Python, but that doesn't bother me and your AI formula assistant is supposed to fill in Python knowledge gaps anyway.

**TL;DR** I definitely prefer your indexed syntax to the mixed terminology you have today. I think I would prefer a separate singular method (with consistent terminology) even more, but I don't feel I have a winning argument for it since it doesn't actually behave as I expect.
